roles/test-utils/mining-device/src/main.rs

```diff
@@ -669,3 +686,74 @@ fn generate_random_32_byte_array() -> [u8; 32] {
     rng.fill(&mut arr[..]);
     arr
 }
+
+fn start_mining_threads(
+    have_new_job: Receiver<()>,
```

Comment:
this receiver is not being triggered only on the arrival of new jobs

every time `Device` receives any message from upstream (e.g.: `SubmitSharesSuccess`), that is also